### PR TITLE
[mail] Correctly named STARTTLS and a few small improvements

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/org.openhab.binding.mail/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -40,22 +41,22 @@
 	</thing-type>
 	<thing-type id="imap" extensible="mailcount">
 		<label>IMAP Server</label>
-		<description>Used for receiving mail</description>
+		<description>Used for receiving mails</description>
```
Comment (cpmeister) [inline bundles/org.openhab.binding.mail/src/main/resources/ESH-INF/thing/thing-types.xml:44]: I think "mail" sounds better than "mails".

**Comment by J-N-K** [comment]
I agree that this is technically correct. But is it worth a breaking change?

File: bundles/org.openhab.binding.mail/src/main/resources/ESH-INF/thing/thing-types.xml (hunk 1 above)
Comment (kaikreuzer) [reply bundles/org.openhab.binding.mail/src/main/resources/ESH-INF/thing/thing-types.xml]: Thanks - I've updated the texts accordingly.

**Comment by kaikreuzer** [comment]
> But is it worth a breaking change?

I was wondering about this as well. But my conclusion was that "TLS" is simply wrong and that's also how I came to this PR: I configured my SMTP server, which uses TLS, so I chose TLS as the option and then wondered why it wasn't working (as it tried to use port 25).
I actually think hardly anybody uses STARTTLS nowadays, so the change shouldn't have a real impact. I kept "SSL" as the technical identifier for the "SSL/TLS" in place, so that this stays fully compatible.

**Comment by cweitkamp** [comment]
I any case we should mention it in the [update.lst](https://github.com/openhab/openhab-distro/blob/2.5.x/distributions/openhab/src/main/resources/bin/update.lst). Shouldn't we?

**Comment by kaikreuzer** [comment]
> I any case we should mention it in the update.lst. Shouldn't we?

Sure, I only wanted to wait with such an PR until the change gets accepted. Here we go: https://github.com/openhab/openhab-distro/pull/1099

**Pull request merged**